core/cmd/local_client.go

```diff
@@ -73,6 +75,24 @@ func loggedStop(app chainlink.Application) {
 	logger.WarnIf(app.Stop())
 }
 
+func checkFilePermissions(directory string) {
+	invalidPermissions := os.FileMode(0077)
+	err := filepath.Walk(directory,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			m := info.Mode()
+			if m&invalidPermissions != 0 {
+				return fmt.Errorf("%s has incorrect file permissions, %s", path, m.String())
+			}
+			return nil
+		})
+	if err != nil {
+		logger.Fatal(err)
+	}
```

Comment:
It looks like early exit on error is handled in the caller, maybe we could do the same?  It would also make the function easier to test